docs/concepts/rewriters.md

```diff
@@ -0,0 +1,565 @@
+
+- `Initial`
+
+    The initial form of the rewriter instance.
+
+- `Simplify`
+
+    A backend-specific 'simplify' command.
+
+- `Expand`
+
+    Expand all brackets in the expression.
+
+- `Assumption`
+
+    Add an assumption onto a symbol.
+
+- `Substitution`
+
+    Substitute a symbol or expression for another.
+
+- `ReapplyAllAssumptions`
+
+    Reapply all previously applied assumptions.
+
+The primary purpose of `Instructions` is to track the history of an expression, and the user may only ever interact with these objects via the `history()` method. Of these classes, only `Assumption` and `Substitution` require special logic; the others have no methods or attributes implemented. We explore `Assumption` and `Substitution` in more detail below.
+
+!!! note "Why not just an `Enum`?"  
+    Originally the `Instructions` were implemented as an `Enum`! However, because `Assumptions` and `Substitutions` required special logic it was challenging to enforce strict typing across both an `Enum` and other dataclasses. Creating an empty class `Instruction` that everything else subclassed from is a cleaner implementation.
+
+
+#### Assumptions
+Assumptions about symbols can be input to the expression, and (in the case of SymPy) the backend symbolic engine attempts to simplify the expression with this new knowledge. An assumption requires three arguments:
+
+- `symbol_name: str`
+
+    The symbol to add the assumption to.
+
+- `comparator: Comparator | str`
+
+    The comparison to apply, one of ">", "<", ">=", "<=". 
+
+- `value: int | float`
+
+    The reference value to compare the symbol to.
+
+Alternatively an assumption can be parsed directly from a string:
+```python
+sympy_rewriter("max(0, X)").assume("X > 0")
+>>> X
+```
+
+Given an input assumption to a `sympy_rewriter`, the symbol is updated with the relevant SymPy [predicates](https://docs.sympy.org/latest/guides/assumptions.html#predicates). For some symbol `X` the predicates we support are:
+
+- positive: `X > a > 0` or `X > 0`,
+- nonnegative: `X >= 0` or `positive`,
+- negative: `X < a < 0` or `X < 0`,
+- nonpositive: `X<=0` or `negative`,
+
+where `a` is some real number. From these SymPy is able to deduce other predicates. We do not implement predicates that declare if a symbol belongs to a particular number group, i.e. `integer`, `complex`, etc. We found that these kinds of assumptions did little to help simplify expressions. Similarly we do not have support for symbols being declared as infinitely large; if there is a valuable use case for these then they are easily added.
+
+
+There is unfortunately [no way to input assumptions between different symbols](https://docs.sympy.org/latest/guides/assumptions.html#relations-between-different-symbols). Similarly SymPy does not implement predicates that specify a relationship between a symbol and some nonzero value, i.e. `X > 5`. However, for this latter point, we have implemented a workaround. 
```

Comment:
It's great that we have a a very specific link to sympy docs which show that it's not possible. Do we also give people this link when we throw a warning in that method? Could be a good idea.